src/charm.py

```diff
@@ -379,6 +394,53 @@ def _pebble_layer(self):
         return pebble_layer
 
     @property
+    def self_probe(self):
+        """The self-monitoring blackbox probe."""
+        probe = {
+            "job_name": "blackbox_http_2xx",
+            "params": {"module": ["http_2xx"]},
+            "static_configs": [
+                {
+                    "targets": [self.external_url + "/api/v1/health/"],
+                    "labels": {"name": "cos-registration-server"},
+                }
+            ],
+        }
+        return [probe]
+
+    @property
+    def devices_ip_endpoints_probes(self):
+        """The devices IPs from the server database."""
+        database_url = (
+            self.external_url
+            + COS_REGISTRATION_SERVER_API_URL_BASE
+            + "devices/?fields=uid,address"
+        )
+        devices_addresses = []
+        devices_uids = []
+        try:
+            response = requests.get(database_url)
+            response.raise_for_status()
+            response_json = response.json()
+            if response_json:
```

Comment:
Should there be an `else`? If, for any reason, this is false, `devices_addresses` and `devices_uids` will be used undefined in the following for-loop.

Reply:
mm but i'm defining them before the try with `devices_addresses = []` and `devices_uids = []`, so if for any reason the response_json is None, it will proceed to the for loop, but zip will skip it and the function will just return jobs = [].

Reply:
You're right, I missed that :+1: .